src/C++/SSLSocketInitiator.cpp

```diff
@@ -318,83 +323,163 @@ void SSLSocketInitiator::doConnect( const SessionID& s, const Dictionary& d )
     setPending( s );
     m_pendingConnections[ result ] = new SSLSocketConnection( *this, s, result, ssl, &m_connector.getMonitor() );
   }
-  catch ( std::exception& ) {}
+  catch ( std::exception& ex)
+  {
+      getLog()->onEvent(ex.what());
+  }
 }
 
-bool SSLSocketInitiator::handshakeSSL(SSL* ssl)
+SSLHandshakeStatus SSLSocketInitiator::handshakeSSL(SSLSocketConnection* connection)
 {
+    SSL* ssl = connection->sslObject();
     ERR_clear_error();
     // Do the SSL handshake.
     int rc = SSL_connect(ssl);
-    while (rc <= 0) {
+    if (rc <= 0) {
```

Comment:
We get rid of the spin loop that kept retrying until the socket read/write data.
This caused an issue because if the socket stayed connected but didn't send data the code would spin forever.
The method will now have to be called again from onRead/onWrite to resume the OpenSSL handshake's state-machine.